docs/harbor_backup_restore.md

```diff
@@ -0,0 +1,247 @@
+This will create a namespace called `velero`, and place a deployment named `velero` in it.
+Note that the installation command sets the bucket `velero-backup` that has been created a few steps 
+earlier as well as EC2 credentials located in `~/.aws/credentials` file. Also note that 
+the `region` and `s3Url` parameters are GX-SCS specific. For further details about installation
+options, supported storage providers, and more visit the official Velero [docs](https://velero.io/docs/).
+
+- If you want to use `snapshot` to back up Harbor data:
+  - Install Velero:
+  ```bash
+   velero install \
+     --features=EnableCSI \
+     --provider aws \
+     --plugins velero/velero-plugin-for-aws:v1.6.1,velero/velero-plugin-for-csi:v0.4.2 \
+     --bucket velero-backup \
+     --secret-file ~/.aws/credentials \
+     --backup-location-config region=RegionOne,s3ForcePathStyle="true",s3Url=https://api.gx-scs.sovereignit.cloud:8080 \
+     --snapshot-location-config region=RegionOne,enableSharedConfig=true
+   ```
+   - In order to allow Velero to do Volume Snapshots, we need to deploy a new VolumeSnapshotClass. Create a velero-snapclass.yaml file as follows:
+  ```bash
+  cat > velero-snapclass.yaml << EOF
+  apiVersion: snapshot.storage.k8s.io/v1
+  deletionPolicy: Delete
+  driver: cinder.csi.openstack.org
+  kind: VolumeSnapshotClass
+  metadata:
+    name: csi-cinder-snapclass-in-use-v1-velero
+    labels:
+      velero.io/csi-volumesnapshot-class: "true"
+  parameters:
+    force-create: "true"
+  EOF
+  ```
+  - Apply the new class:
+  ```bash
+  kubectl apply -f velero-snapclass.yaml
+  ```
+- If you want to create Harbor `backup` with Restic: 
+  - Install Velero:
+  ```bash
+  velero install \
+    --provider aws \
+    --plugins velero/velero-plugin-for-aws:v1.6.1 \
+    --bucket velero-backup \
+    --secret-file ~/.aws/credentials \
+    --use-volume-snapshots=false \
+    --uploader-type=restic \
+    --use-node-agent \
+    --backup-location-config region=RegionOne,s3ForcePathStyle="true",s3Url=https://api.gx-scs.sovereignit.cloud:8080
+  ```
+
+##  Backup and restore
+
+Note that the following backup steps mainly point to actions from an official [Backup And Restore Harbor With Velero](https://goharbor.io/docs/main/administration/backup-restore) tutorial.
+In this guide, find the added value from additional explanations/hints and up-to-date commands.
+
+Harbor, by design, consists of multiple (micro)services that could store their data
+variously, based on the Harbor configuration. See the [Harbor persistence](harbor_persistence.md)
+docs for further information regarding the Harbor persistence layer. The following steps cover
+cases when Harbor persistence is enabled and the "internal" databases (PostgreSQL and Redis)
+are used.
+
+Note that backups of external databases (PostgreSQL and Redis) are not supported by the 
+official tutorial and are out of the scope of this guide as well.
```

Comment:
This left my wondering if backing up postgres and redis is required at all. Does it only hold barely important metadata or is it mandatory for restoring?

Please consider clarifying this explicitly on the guide.

Reply:
Explained in more detail within fixup f5896e66a94456185911801b6a66b1ac790c73cc